test/spec/features/append/AppendNodeProvider.spec.js

```diff
@@ -0,0 +1,116 @@
+import {
+  insertCoreStyles,
+  insertBpmnStyles,
+  bootstrapModeler,
+  inject
+} from 'test/TestHelper';
+
+import {
+  query as domQuery
+} from 'min-dom';
+
+import {
+  forEach
+} from 'min-dash';
+
+import Append from 'lib/features/append';
+
+import {
+  CreateAppendAnythingModule
+} from 'bpmn-js-create-append-anything';
+
+import diagramXML from './AppendNode.bpmn';
+
+insertCoreStyles();
+insertBpmnStyles();
+
+describe('<AppendNodeProvider>', function() {
+
+  function bootstrap(diagramXML) {
+
+    return function() {
+      return bootstrapModeler(diagramXML, {
+        additionalModules: [
+          Append,
+          CreateAppendAnythingModule
+        ]
+      })();
+    };
+  }
+
+  beforeEach(bootstrap(diagramXML));
+
+
+  describe('should trigger action', function() {
+
+    it('click', inject(function(elementRegistry, appendNode) {
+
+      // given
+      const shape = elementRegistry.get('Task');
+
+      // when
+      appendNode.show(shape);
+      appendNode.trigger('click', shape);
+
+      // then
+      expect(domQuery('.djs-popup.bpmn-append')).to.exist;
+    }));
+
+    it('dragstart', inject(function(elementRegistry, appendNode, dragging) {
+
+      // given
+      const shape = elementRegistry.get('Task');
+
+      // when
+      appendNode.show(shape);
+      appendNode.trigger('dragstart', shape);
+
+      // then
+      expect(dragging.context().data.context.start).to.eql(shape);
+    }));
+
+  });
+
+
+  describe('#isAllowed', function() {
+
+    it('should return true', inject(function(elementRegistry, appendNodeProvider) {
+
+      // given
+      const shape = elementRegistry.get('Task');
+
+      // when
+      const allowed = appendNodeProvider.isAllowed(shape);
+
+      // then
+      expect(allowed).to.be.true;
+    }));
+
+
+    it('should return false', inject(function(elementRegistry, appendNodeProvider) {
+
+      // given
+      const ids = [
+        'EndEvent',
+        'Group',
+        'TextAnnotation',
+        'Lane',
+        'Participant',
+        'DataStoreReference',
+        'DataObjectReference'
+      ];
+
+      // when
+      forEach(ids, function(id) {
+        const shape = elementRegistry.get(id);
+
+        const allowed = appendNodeProvider.isAllowed(shape);
+
+        // then
+        expect(allowed).to.be.false;
```

Comment:
Note that if this ever breaks, we will only see `expected true to be false`. Perhaps it could be a more future-proof solution to place `it` inside of iteration, so that the test name takes form of "it should disallow append node for {elementId}"